extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -128,6 +132,29 @@ export async function runRemoteQuery(cliServer: cli.CodeQLCliServer, credentials
     void showAndLogInformationMessage(`Successfully scheduled runs. [Click here to see the progress](https://github.com/${OWNER}/${REPO}/actions).`);
 
   } catch (error) {
-    void showAndLogErrorMessage(error);
+    if (typeof error.message === 'string' && error.message.includes('Some repositories were invalid')) {
+      const invalidRepos = error?.response?.data?.invalid_repos || [];
+      const reposWithoutDbUploads = error?.response?.data?.repos_without_db_uploads || [];
+      void logger.log('Unable to run query on all repositories');
+      void logger.log('invalidRepos = ' + JSON.stringify(invalidRepos));
+      void logger.log('reposWithoutDbUploads = ' + JSON.stringify(reposWithoutDbUploads));
+
+      if (invalidRepos.length + reposWithoutDbUploads.length === repositories.length) {
+        // Every repo is invalid in some way
+        void showAndLogErrorMessage('Unable to run query on the specified repositories.');
+        return;
+      }
+
+      const popupMessage = 'Unable to run query on the specified repositories. See logs for more details.';
```

Comment:
@aeisenberg (or anyone else!) do you know if there's a way to link to the Extension Log from a message? It would be neat if "See logs" was a clickable link 🖱️

Reply:
We do this in `internalShowAndLog`, added as a button to the dialog box. In theory, you can add multiple buttons and handle different actions for each button (though that's not _exactly_ what you are asking).

https://github.com/github/vscode-codeql/blob/7e78a6bc5c852908e5284eb71ed15170a0c28156/extensions/ql-vscode/src/helpers.ts#L72-L72

I think markdown is accepted in these dialog boxes, so you _might_ be able to use a [command uri(https://code.visualstudio.com/api/extension-guides/command#command-uris). But, to do that I thnk you will need to create a command that will open the extension log (I don't think there's a builtin command that does it).

Reply:
```suggestion
      const popupMessage = 'Unable to run query on some of the specified repositories. See logs for more details.';
```

Reply:
The problem we were having with `internalShowAndLog` and similar methods is that clicking any of the options closes the message. It would be nice if you could click to see more info and then come back to decide whether to re-run your query or not.

The command URIs look pretty interesting and could do what we want. Still haven't quite decided what that is though.